pkg/csi/cinder/controllerserver.go

```diff
@@ -56,18 +56,30 @@ const (
 func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
 	klog.V(4).Infof("CreateVolume: called with args %+v", protosanitizer.StripSecrets(req))
 
-	// Volume cloud
-	volCloud := req.GetSecrets()["cloud"]
-	cloud, cloudExist := cs.Clouds[volCloud]
-	if !cloudExist {
-		return nil, status.Error(codes.InvalidArgument, "[CreateVolume] specified cloud undefined")
-	}
-
 	// Volume Name
 	volName := req.GetName()
 	volCapabilities := req.GetVolumeCapabilities()
 	volParams := req.GetParameters()
 
+	// Volume Type
+	volType := volParams["type"]
+
+	var volAvailability string
+	var volCloudAvailability string
+	if cs.Driver.withTopology {
+		// First check if volAvailability is already specified, if not get preferred from Topology
+		// Required, incase vol AZ is different from node AZ
+		volAvailability = volParams["availability"]
+		if volAvailability == "" {
+			accessibleTopologyReq := req.GetAccessibilityRequirements()
+			// Check from Topology
+			if accessibleTopologyReq != nil {
+				volCloudAvailability = sharedcsi.GetAZFromTopology(withTopologyKey, accessibleTopologyReq) // = region defined from the node label topology.kubernetes.io/region=region
```

Comment:
I don't like reusing this method to fetch a fundamentally different value (region rather than zone). If nothing else, the logging is going to be very confusing. Would it be possible to rename and rework this function so it retrieves both the zone and region? I suspect we can just ignore the latter for manila for now.

Also, the comments on both lines are confusing. Can you remove them?